src/js/tracing/stalltracking.ts

```diff
@@ -215,6 +226,21 @@ export class StallTrackingInstrumentation {
     );
   }
 
+  /**
+   * Switch that enables the iteraction once app moves from background to foreground.
+   */
+  private _backgroundEventListener(state: AppStateStatus): void {
+    if (state === ('active' as AppStateStatus)) {
+      this._isBackground = false;
+      if (this._timeout != null) {
+        this._lastIntervalMs = timestampInSeconds() * 1000;
+        this._iteration();
+      }
+    } else {
+      this._isBackground = true;
```

Comment:
H: Should we clear the timeout here to avoid a possible race condition where we set `isBackground` but the app gets paused before the `_iteration` gets executed?

```suggestion
      this._isBackground = true;
      this._timeout !== null && clearTimeout(this._timeout);
```

Reply:
I have appleid your suggestion on another commit so I'll be resolving this one.